scripts/controllers/RobotArm.py

```diff
@@ -0,0 +1,198 @@
+            self.vel_pubs[index].publish(Float64(vel))
+
+    def set_joint_trajectory(self, trajectories, time_per_step=1.0):
+        """
+        sends a joint trajectory command.
+        """
+        if not rospy.is_shutdown():
+            self.controller_manager.switch_mode(ControllerType.TRAJECTORY)
+
+            # 3d tensor of shape (3, n_joint, n_point)
+            points_tensor = np.array(trajectories)
+            assert points_tensor.shape[1] == 3, "3 row should be provided for pos, vel and acc."
+            assert len(points_tensor.shape) == 3, "Trajectories should be of 3-d shape!"
+            assert points_tensor.shape[2] == self.num_joints, "Trajectory should " \
+                                                              "account for num joints."
+
+            num_points = points_tensor.shape[0]
+            time_from_start = time_per_step
+            time_from_start_duration = rospy.Duration(time_from_start)
+            msg = JointTrajectory()
+            msg.points = []
+            for i in range(num_points):
+                point = JointTrajectoryPoint()
+                point.positions = list(points_tensor[i, 0])
+                point.velocities = list(points_tensor[i, 1])
+                point.accelerations = list(points_tensor[i, 2])
+                point.time_from_start = time_from_start_duration
+                time_from_start += time_per_step
+                time_from_start_duration = rospy.Duration(time_from_start)
+                msg.points.append(point)
+            msg.header.stamp = rospy.Time.now()
+            msg.header.frame_id = '/base_link'
+            msg.joint_names = list(self.names)
+
+            self.traj_pub.publish(msg)
+
+    def joint_state_callback(self, data):
+        """
+        callback that takes care of the joint state data.
+        """
+        self.joint_data = data
+        if self.names is None:
+            names_arr = np.array(self.joint_data.name)
+            arr = np.isin(names_arr, self.ignore_joints_arr)
+            indices = np.squeeze(np.argwhere(arr == False))  # noqa: E712
+            self.valid_indices = indices
+            self.names = names_arr[self.valid_indices]
+            self.mapping = {v: i for i, v in enumerate(self.names)}
+            self.positions = {name: 0.0 for name in self.names}
+
+            # construct mapping on joint names to indices
+        self.data_exists = True
+
+    def get_publishers(self, controller_names):
+        """
+        gets position, velocity, and trajectory publisher(s). Lengths may
+        be different.
+        """
+        joint_position_pubs = []
+        joint_velocity_pubs = []
+        pos_controller_names = controller_names[0]
+        vel_controller_names = controller_names[1]
+        traj_controller_name = controller_names[2][0]
+        traj_name = '/{}/command'.format(traj_controller_name)
+        traj_pub = rospy.Publisher(traj_name, JointTrajectory, queue_size=1)
```

Comment:
What if you did not have `traj_controller`???
Or does `RobotArm` need to be inherited if someone wanted to add a new robot arm?

Reply:
Wouldn't it be better to name the `RobotArm` as `PandaArm` then?

Reply:
Initially I was thinking that (as this class is used for Panda), but I wanted to keep it with the generic name in the case that we add new robot arms.